catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/CatalogFrameworkImplTest.java

```diff
@@ -2354,6 +2358,69 @@ public void testFederatedQueryPermissions() throws Exception {
                 .size(), is(1));
     }
 
+    @Test
+    public void testNoFanoutExcludesNoTagsProperty() throws Exception {
+        Consumer<FederationStrategy> runner = federationStrategy -> {
+            QueryRequest request = mock(QueryRequest.class);
+            HashMap<String, Serializable> requestProps = new HashMap<>();
+            when(request.getProperties()).thenReturn(requestProps);
+
+            when(request.getQuery()).thenReturn(mock(Query.class));
+            QueryResponse response = mock(QueryResponse.class);
+            try {
+                when(federationStrategy.federate(anyListOf(Source.class),
+                        any(QueryRequest.class))).thenReturn(response);
+                framework.query(request);
+            } catch (FederationException | UnsupportedQueryException e) {
+                throw new RuntimeException(e);
+            }
+            assertThat(requestProps.containsKey("no-default-tags"), is(Boolean.FALSE));
+        };
+
+        noTagsPropertyTest(runner);
+    }
+
+    @Test
+    public void testFanoutHasNoTagsProperty() throws Exception {
+        Consumer<FederationStrategy> runner = federationStrategy -> {
+            QueryRequest request = mock(QueryRequest.class);
+            HashMap<String, Serializable> requestProps = new HashMap<>();
+            when(request.getProperties()).thenReturn(requestProps);
+
+            when(request.getQuery()).thenReturn(mock(Query.class));
+            framework.setFanoutEnabled(true);
```

Comment:
Java is not great with arbitrary numbers of arguments to lambdas, but I'll see if I can't simplify this a bit without needing to nest `BiFunction`s.